public/app/features/explore/state/query.ts

```diff
@@ -323,7 +323,8 @@ async function handleHistory(
   // Because filtering happens in the backend we cannot add a new entry without checking if it matches currently
   // used filters. Instead, we refresh the query history list.
   // TODO: run only if Query History list is opened (#47252)
-  dispatch(loadRichHistory(exploreId));
+  await dispatch(loadRichHistory(ExploreId.left));
```

Comment:
Just wondering why we only seem to have logic for one exploreId/side of the split above but re reload the history on both sides?

Reply:
This is because this method handles adding a query run from a single pane (left OR right), but depending on applied filters, adding a new query in the history may affect the query history list in both panes (left AND right). 

This is actually something I wonder we could even get rid of. It covers only cases when a user runs a query with query history already opened. It had worked before so I added it to keep the behavior, but to be honest, I'm not sure how important it is and it makes the code more complex.